### ISSUE TITLE
PR185 Seems to not actually work, as FirestoreRules are very finicky

### ISSUE THREAD

**Issue by nazrinharris** [nazrinharris/farmhub#195]
Though it seems like it worked, I have recently discovered that this in fact, does not work. Mainly due to the limitations of `String` comparisons. To work around this, we would need to totally rethink how we view app versions. Basically, we want the semantic versioning, `0.3.2`, to be represented as simple comparable integers. Here's some info:

`semantic_app_version` - `0.3.2`
`integer_app_version` - `302`

Where the formula is as such:
`integer_app_version = major * 10000 + minor * 100 + patch`

This way, comparisons are simple and straightforward. The interfaces I (dev) am interested when it comes to versioning, such as the iOS and Android app version, minimum and latest app version in remote config will all be semantic, but whenever comparisons need to be made, we will convert to its integer representation.

This would be in a new PR, after #192 completes. Basically #192 compares directly client-side, and doesn't deal with Firestore Rules, so it can go ahead like usual. But we can consider using integer comparison to unify comparisons.

TLDR;
PR185 didn't actually work. #192 will work and make sure the "first layer of defense" work. A new PR will be made for this "second layer", where Firestore itself will decline writes on outdated apps, rather than a client-check.
